### PR TITLE
chore(query): refactor agg state merge

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13153)

### CODE REVIEW

**Comment by sundy-li** [comment]
askbend:summary

File: src/query/expression/src/utils/block_debug.rs [hunk 1]
```diff
@@ -177,46 +177,17 @@ fn create_box_table(
...
-        for row in 0..top_rows {
-            let mut v = vec![];
-            for block_entry in top_collection.columns() {
-                let value = block_entry.value.index(row).unwrap().to_string();
-                if replace_newline {
-                    v.push(value.to_string().replace('\n', "\\n"));
-                } else {
-                    v.push(value.to_string());
-                }
+
```
Comment (RinChanNOWWW) [inline src/query/expression/src/utils/block_debug.rs:180]: How about adding a unit test for this function?

File: src/query/functions/src/aggregates/adaptors/aggregate_ornull_adaptor.rs [hunk 2]
```diff
@@ -173,16 +173,16 @@ impl AggregateFunction for AggregateFunctionOrNullAdaptor {
     }
 
     #[inline]
-    fn deserialize(&self, place: StateAddr, reader: &mut &[u8]) -> Result<()> {
-        let flag = reader[reader.len() - 1];
```
Comment (RinChanNOWWW) [inline src/query/functions/src/aggregates/adaptors/aggregate_ornull_adaptor.rs:177]: The old logic is wrong, is there any case can produce the wrong result?

**Review by zhang2014 — APPROVED** [review]

**Review by RinChanNOWWW — CHANGES_REQUESTED** [review]
The `arena_holder` in `HashTablePayload` can also be removed.

**Review by RinChanNOWWW — COMMENTED** [review]
Others LGTM.

File: src/query/functions/src/aggregates/adaptors/aggregate_ornull_adaptor.rs (hunk 2 above)
Comment (sundy-li) [reply src/query/functions/src/aggregates/adaptors/aggregate_ornull_adaptor.rs:177]: the old logic is correct, the new `merge` is  `old deserialize` + `old merge`

**Review by RinChanNOWWW — APPROVED** [review]

**Pull request merged**